src/FSharp.Formatting.Markdown/MarkdownParser.fs

```diff
@@ -69,17 +69,127 @@ let inline (|EscapedLatexInlineMathChar|_|) input =
 
-    if match'.Success then
-        let entity = match'.Value
-        let _, rest = List.splitAt entity.Length input
-        Some(char entity, rest)
-    else
-        None
+        let match' = Regex.Match(Array.ofList input |> String, re)
+
+        if match'.Success then
+            let entity = match'.Value
+            let _, rest = List.splitAt entity.Length input
+            Some(char entity, rest)
+        else
+            None
+
+let inline (|NotPunctuation|_|) input =
+    match input with
+    | Punctuation _ -> None
+    | _ -> Some input
+
+module Char =
+    let inline (|WhiteSpace|_|) input =
+        match input with
+        | [] -> Some input
+        | x :: _xs ->
+            if String.IsNullOrWhiteSpace(string x) then
+                Some input
+            else
+                None
+
+    let inline (|NotWhiteSpace|_|) input =
+        match input with
+        | WhiteSpace _ -> None
+        | _ -> Some input
+
+/// Succeeds when the specificed character list starts with a delimeter run.
+let inline (|DelimiterRun|_|) input =
+    match input with
+    | ('*'
+    | '_') :: _tail as (h :: t) ->
+        let run, rest = List.partitionWhile (fun x -> x = h) (h :: t)
+        Some(run, rest)
+    | _ -> None
+
+/// Succeeds when there's a match to a string of * or _ that could
+/// open emphasis.
+let inline (|LeftDelimiterRun|_|) input =
```

Comment:
I'll remove the inline - these are too big to inline and there will be no useful performance gain from using it